flink-python/src/test/java/org/apache/flink/table/runtime/operators/python/aggregate/PythonStreamGroupWindowAggregateOperatorTest.java

```diff
@@ -0,0 +1,1129 @@
+
+        testHarness.close();
+    }
+
+    @Override
+    public LogicalType[] getOutputLogicalType() {
+        return new LogicalType[] {
+            DataTypes.STRING().getLogicalType(), DataTypes.BIGINT().getLogicalType()
+        };
+    }
+
+    @Override
+    public RowType getInputType() {
+        return new RowType(
+                Arrays.asList(
+                        new RowType.RowField("f1", new VarCharType()),
+                        new RowType.RowField("f2", new VarCharType()),
+                        new RowType.RowField("f3", new BigIntType()),
+                        new RowType.RowField("rowTime", new BigIntType())));
+    }
+
+    @Override
+    public RowType getOutputType() {
+        return new RowType(
+                Arrays.asList(
+                        new RowType.RowField("f1", new VarCharType()),
+                        new RowType.RowField("f2", new BigIntType()),
+                        new RowType.RowField("windowStart", new TimestampType(3)),
+                        new RowType.RowField("windowEnd", new TimestampType(3))));
+    }
+
+    @Override
+    OneInputStreamOperator getTestOperator(Configuration config) {
+        long size = 10000L;
+        long slide = 5000L;
+        SlidingWindowAssigner windowAssigner =
+                SlidingWindowAssigner.of(Duration.ofMillis(size), Duration.ofMillis(slide))
+                        .withEventTime();
+        PlannerWindowReference windowRef =
+                new PlannerWindowReference("w$", new Some<>(new TimestampType(3)));
+        LogicalWindow window =
+                new SlidingGroupWindow(
+                        windowRef,
+                        new FieldReferenceExpression(
+                                "rowtime",
+                                new AtomicDataType(
+                                        new TimestampType(true, TimestampKind.ROWTIME, 3)),
+                                0,
+                                3),
+                        intervalOfMillis(size),
+                        intervalOfMillis(slide));
+        return new PassThroughPythonStreamGroupWindowAggregateOperator(
+                config,
+                getInputType(),
+                getOutputType(),
+                new PythonAggregateFunctionInfo[] {
+                    new PythonAggregateFunctionInfo(
+                            PythonScalarFunctionOperatorTestBase.DummyPythonFunction.INSTANCE,
+                            new Integer[] {2},
+                            -1,
+                            false)
+                },
+                getGrouping(),
+                -1,
+                false,
+                false,
+                3,
+                windowAssigner,
+                window,
+                0,
+                new int[] {0, 1});
+    }
+
+    /** PassThroughPythonStreamGroupWindowAggregateOperator. */
+    public static class PassThroughPythonStreamGroupWindowAggregateOperator<K>
```

Comment:
It would be better if we extract this class to a independent java file. It is so large.

Reply:
Yes, it is really bad for reading